include/revng/Pipes/TaggedFunctionKind.h

```diff
@@ -20,6 +20,16 @@ class TaggedFunctionKind : public pipeline::LLVMKind {
 private:
   const FunctionTags::Tag *Tag;
 
+  // It is on the heap to avoid Initialization Order Fiasco
+  std::unique_ptr<llvm::SmallVector<TaggedFunctionKind *>> Children = nullptr;
+
+  void registerChild(TaggedFunctionKind *Child) {
+    if (Children == nullptr)
+      Children = std::make_unique<llvm::SmallVector<TaggedFunctionKind *>>();
+
+    Children->push_back(Child);
+  }
```

Comment:
* This commit introduces this function but does not use it.
* I'm confused why we need `children`. `Kind` can already enumerate its children.

Reply:
the issue is that the childs in kind are of type Kind and here we need them of type TaggedFunctionKind, and i have not implemented the required stuff to make them upcastable with llvm::cast